src/utils/resolve.ts

```diff
@@ -162,33 +165,55 @@ function fullResolve(
       fileExistsCache.set(cacheKey, resolved.path as string | null)
       return resolved
     }
-  } else {
-    const configResolvers = settings['import-x/resolver-legacy'] ||
-      settings['import-x/resolver'] || {
-        node: settings['import-x/resolve'],
-      } // backward compatibility
-
-    for (const { enable, options, resolver } of normalizeConfigResolvers(
-      configResolvers,
+  } else if (
+    // backward compatibility for very old `import-x/resolve` options that is no longer supported
+    Object.prototype.hasOwnProperty.call(settings, 'import-x/resolve') &&
+    settings['import-x/resolve']
+  ) {
+    nodeResolverInstanceForLegacyNodeResolverSettings ||= createNodeResolver({
+      extensions: (settings['import-x/resolve'].extensions ||
```

Comment:
Extract `settings['import-x/resolve']` as single variable?

Reply:
Done in [`7dab322` (#272)](https://github.com/un-ts/eslint-plugin-import-x/pull/272/commits/7dab322c2fc41bbf101f249acf0549aab1c15e2b)